### PR TITLE
Allow Symfony 6

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,14 +12,14 @@
     "minimum-stability": "dev",
     "require": {
         "php": ">=7.1",
-        "composer-plugin-api": "^1.0|^2.0"
+        "composer-plugin-api": ">=1.0"
```
Comment (stof) [inline composer.json:15]: Do we actually want to do this ? This seems dangerous regarding changes in the composer plugin API (and using an incompatible plugin in composer can break you ability to upgrade...)

File: tests/UnpackerTest.php [hunk 2]
```diff
@@ -40,6 +40,9 @@ public function testDoNotDuplicateEntry(): void
         file_put_contents($composerJsonPath, '{}');
 
         $originalEnvComposer = getenv('COMPOSER');
+        $originalEnvComposer = $_SERVER['COMPOSER'];
```
Comment (stof) [inline tests/UnpackerTest.php:43]: should it be a different variable name ?

File: composer.json (hunk 1 above)
Comment (nicolas-grekas) [reply composer.json:15]: ok, reverting, we might decide to widen later on if needed

File: tests/UnpackerTest.php (hunk 2 above)
Comment (nicolas-grekas) [reply tests/UnpackerTest.php:43]: no need to, esp for a test file

File: composer.json (hunk 1 above)
Comment (nicolas-grekas) [reply composer.json:15]: Reverted in https://github.com/symfony/flex/commit/e2ea724773cc7c391e0d163967675acd31ee04c7

File: tests/UnpackerTest.php (hunk 2 above)
Comment (stof) [reply tests/UnpackerTest.php:43]: then the previous line is dead code.

**Pull request merged**